sdcm/utils/docker_utils.py

```diff
@@ -399,18 +401,29 @@ def build_container_image(cls, instance: INodeWithContainerManager, name: str, *
             getattr(instance, f"{name.family}_container_image_build_args", cls._build_args)(**extra_build_args))
 
         LOGGER.debug("Build arguments for Docker image `%s':\n%s,", image_tag, pformat(build_args, indent=8))
-        try:
-            image, logs = docker_client.images.build(tag=image_tag, **dockerfile_args, **build_args)
-        except BuildError as docker_build_error:
-            LOGGER.critical("Build log:\n%s",
-                            "\n".join(item['stream'] for item in docker_build_error.build_log if "stream" in item))
-            raise
-
-        LOGGER.debug(">>> Build log for Docker image `%s': >>>", image_tag)
-        for entry in logs:
-            if "stream" in entry:
-                LOGGER.debug(entry["stream"].rstrip())
-        LOGGER.debug("<<<")
+
+        resp = docker_client.api.build(tag=image_tag, **dockerfile_args, **build_args)
+        last_event = None
+        image_id = None
+        LOGGER.info(">>> Build log for Docker image `%s': >>>", image_tag)
+        result_stream, internal_stream = itertools.tee(json_stream(resp))
+        for chunk in internal_stream:
+            if 'error' in chunk:
+                raise BuildError(chunk['error'], result_stream)
+            if 'stream' in chunk:
+                LOGGER.info(chunk["stream"].rstrip())
+                match = re.search(
+                    r'(^Successfully built |sha256:)([0-9a-f]+)$',
+                    chunk['stream']
+                )
+                if match:
+                    image_id = match.group(2)
```

Comment:
Parsing out the image ID from the logs looks fragile.
We don't make any influence on the docker source code and dependency on the output structure may make us fix it periodically for each of the supported SCT branches.

Cannot we make some addition API call after the `build` one finishes?

Reply:
no, this is exactly the code from the `docker.images.build`

Reply:
Since you want to keep it as-is then added comment above.